guacamole-ext/src/main/java/org/apache/guacamole/properties/TimeZoneGuacamoleProperty.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.properties;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.TimeZone;
+import java.util.regex.Pattern;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.GuacamoleServerException;
+
+/**
+ * A GuacamoleProperty whose value is a TimeZone.
+ */
+public abstract class TimeZoneGuacamoleProperty
+        implements GuacamoleProperty<TimeZone> {
+    
+    /**
+     * The set of known good timezone identifiers.
+     */
+    public static final HashSet<String> TZ_IDS =
+            new HashSet<String>(Arrays.asList(TimeZone.getAvailableIDs()));
+    
+    /**
+     * A regex that matches valid variants of GMT timezones.
+     */
+    public static final Pattern GMT_REGEX =
+            Pattern.compile("^GMT([+-](([0]{3,4})|([0]{1,2}([:][0]{2})?)))?$",
+                    Pattern.MULTILINE);
+    
+    @Override
+    public TimeZone parseValue(String value) throws GuacamoleException {
+        
+        // Nothing in, nothing out
+        if (value == null || value.isEmpty())
+            return null;
+        
+        // Attempt to return the TimeZone of the provided string value.
+        TimeZone tz = TimeZone.getTimeZone(value);
+        
+        // If the TimeZone is in the available list, return it.
+        if (TZ_IDS.contains(value))
+            return tz;
+        
+        // If the TimeZone matches the GMT regex, return it.
+        if (GMT_REGEX.matcher(value).matches())
+            return tz;
+        
+        // If the output of the TimeZone matches the input, it's probably valid
+        if (tz.getID().equals(value))
+            return tz;
+        
+        // Nothing valid is found, so throw exception.
+        throw new GuacamoleServerException("Property \"" + getName()
+                + "\" does not specify a valid time zone.");
```

Comment:
I think we should test for invalidity only if the resulting `TimeZone` is not identical to GMT. Any other value returned by `TimeZone.getTimeZone()` would mean that `value` must be valid. That reduces the scope of the problem from "is this timezone valid?" to "did the user really mean GMT?".

Reply:
Makes sense.  Implemented as so.